metadata/src/main/java/org/apache/kafka/controller/ClusterControlManager.java

```diff
@@ -336,10 +350,10 @@ public ControllerResult<BrokerRegistrationReply> registerBroker(
                 ", but got cluster ID " + request.clusterId());
         }
         int brokerId = request.brokerId();
+        List<ApiMessageAndVersion> records = new ArrayList<>();
         BrokerRegistration existing = brokerRegistrations.get(brokerId);
         if (version < 2 || existing == null || request.previousBrokerEpoch() != existing.epoch()) {
-            // TODO(KIP-966): Update the ELR if the broker has an unclean shutdown.
-            log.debug("Received an unclean shutdown request");
```

Comment:
I think we inadvertently lost this log message.

Reply:
Good catch.